arches_lingo/src/arches_lingo/components/scheme/editor/SchemeEditor.vue

```diff
@@ -2,13 +2,39 @@
 import { useGettext } from "vue3-gettext";
 import Button from "primevue/button";
 
+import Tabs from "primevue/tabs";
+import TabList from "primevue/tablist";
+import Tab from "primevue/tab";
+import TabPanels from "primevue/tabpanels";
+import TabPanel from "primevue/tabpanel";
+import SchemeNamespace from "../report/SchemeNamespace.vue";
+import { onBeforeUpdate, onUpdated, ref } from "vue";
+import type { DataComponentMode } from "@/arches_lingo/types";
+
 const { $gettext } = useGettext();
 
 const props = defineProps<{
     editorMax: boolean;
+    activeTab: string;
 }>();
+type sectionTypes = typeof SchemeNamespace;
+const childRefs = ref<Array<sectionTypes>>([]);
+const editMode: DataComponentMode = "edit";
+const schemeComponents = [
+    {
+        component: SchemeNamespace,
+        props: {
+            mode: editMode,
+        },
+        on: {
+            update: onUpdated,
+        },
```

Comment:
these _may_ not need to be part of the component config if we expect every component to have the same props/`on` logic

Reply:
fair, I'll keep it simple till we need otherwise.